code/modules/chemistry/Reagents-Misc.dm

```diff
@@ -221,40 +221,53 @@ datum
 		aranesp
 			name = "aranesp"
 			id = "aranesp"
-			description = "An illegal performance enhancing drug. Side effects might include chest pain, seizures, swelling, headache, fever... ... ..."
+			description = "Darbepoetin alfa, commonly known as Aranesp, is an erythropoietic drug used to increase red-blood cell counts. Safe when used in low amounts, dangerous when used to dope."
 			fluid_r = 120
 			fluid_g = 255
 			fluid_b = 240
 			transparency = 215
+			var/theraputicamount = 5
 			value = 41 // 17 18 6
 			viscosity = 0.4
 
-			on_add()
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_regen"))
-					holder.my_atom:add_stam_mod_regen("aranesp", 15)
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"add_stam_mod_max"))
-					holder.my_atom:add_stam_mod_max("aranesp", 25)
+			on_add(var/mob/M)
+				M = holder?.my_atom
+				if(istype(M) && holder.get_reagent_amount(src.id) > theraputicamount)
+					M.add_stam_mod_regen("aranesp", 15)
+					M.add_stam_mod_max("aranesp", 25)
 				return
 
-			on_remove()
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"remove_stam_mod_regen"))
-					holder.my_atom:remove_stam_mod_regen("aranesp")
-				if (istype(holder) && istype(holder.my_atom) && hascall(holder.my_atom,"remove_stam_mod_max"))
-					holder.my_atom:remove_stam_mod_max("aranesp")
+			on_remove(var/mob/M)
+				M = holder?.my_atom
+				if(istype(M))
+					M.remove_stam_mod_regen("aranesp")
+					M.remove_stam_mod_max("aranesp")
 				return
 
 			on_mob_life(var/mob/M, var/mult = 1)
```

Comment:
This proc already passes a mob, no need to cast and check. You just need the `if (!M) M = holder.my_atom` like previous.

Reply:
I'm confused on this, does that mean I don't need the var/mob/M in  on_add and on_remove?